pkg/lifecycle/lifecycle_provider.go

```diff
@@ -0,0 +1,78 @@
+package lifecycle
+
+import (
+	"sync/atomic"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	ConfigName = "lifecycle-image"
+	ConfigKey  = "image"
+)
+
+type RegistryClient interface {
+	Fetch(keychain authn.Keychain, repoName string) (v1.Image, string, error)
+}
+
+type Provider struct {
+	RegistryClient RegistryClient
+	Keychain       authn.Keychain
+	lifecycleImage atomic.Value
+	callbacks      []func()
+	err            error
+}
+
+func NewProvider(lifecycleImageRef string, client RegistryClient, keychain authn.Keychain) (*Provider, error) {
+	img, _, err := client.Fetch(keychain, lifecycleImageRef)
+	if err != nil {
+		return &Provider{}, errors.Wrap(err, "failed to get lifecycle image")
```

Comment:
Does this error prevent the controller from starting?

Reply:
I thought it would be a good idea, but now I think it should not fail to start